apps/ltn/src/shortcuts.rs

```diff
@@ -166,66 +171,48 @@ struct EntryExit {
     // Really this is a DirectedRoadID, but since the pathfinding request later needs to know
     // lanes, just use this
     lane: LaneID,
-    major_road_name: String,
+    major_road_names: BTreeSet<String>,
 }
 
-fn find_entrances(map: &Map, neighbourhood: &Neighbourhood, cell: &Cell) -> Vec<EntryExit> {
-    let mut entrances = Vec::new();
-    for i in &cell.borders {
-        if let Some(major_road_name) = find_major_road_name(map, neighbourhood, *i) {
-            let mut seen: HashSet<DirectedRoadID> = HashSet::new();
-            for l in map.get_i(*i).get_outgoing_lanes(map, PathConstraints::Car) {
-                let dr = map.get_l(l).get_directed_parent();
-                if !seen.contains(&dr) && cell.roads.contains_key(&dr.road) {
-                    entrances.push(EntryExit {
-                        lane: l,
-                        major_road_name: major_road_name.clone(),
-                    });
-                    seen.insert(dr);
-                }
-            }
-        }
-    }
-    entrances
-}
-
-fn find_exits(map: &Map, neighbourhood: &Neighbourhood, cell: &Cell) -> Vec<EntryExit> {
-    let mut exits = Vec::new();
+fn find_entrances_or_exits(
```

Comment:
Also refactored these two methods; the only difference was using incoming or outgoing lanes